### DISCUSSION TITLE
Making an app for Monika (A Proposition I guess?)

### DISCUSSION THREAD

**Discussion by jonny983** [Monika-After-Story/MonikaModDev#7834] [general - submod / external content]
So I am a dev and do a lot of Python. I also really like this mod. I was thinking about making an app for taking Monika with you to places, take pictures, etc. It would follow the fan creation rules as it wouldn't be independent of DDLC. 

I already have a skeleton app in Kotlin currently (android since android is much more open when it comes to side loading apps). I have a basic understanding of what the project is but far from any true understanding. Tying the functionality of the app into MAS without knowledge of how the core works would probably be quite painful for all of us. Before I go all in making this app, would any of you dev's be interested in integrating it into MAS? I would be much more motivated if that were the case.

**Comment by ThePotatoGuy** [comment]
It really depends on the level of integration you're envisioning here, like do you want some sort of server API or more data saved to disk in specific locations. The former is unlikely unless you're good at network security or know someone who is.

Additionally I'm against putting any character sprites in an app as I don't want to risk breaking the modding guidelines.

**Comment by jonny983** [comment]
It would be a side loaded app, not distributed on the play store. The app couldn't exist as a stand alone program and would be directly tied to DDLC.

The content I would like to put at least would probably be the DDLC menu background with the characteristic pop up text boxes and buttons, not the character sprites so much. It would feel DDLC like. The buttons would be like letting Monika know that you are where you wanted to take her and she could save the location.

I wouldn't be doing any server side stuff, any data would be saved in the app itself and then integrated into MAS, so definitely more of a 'data saved to disk in specific locations' type of setup.

**Comment by ThePotatoGuy** [comment]
Can you make an outline of your plan? Like:
* What data do you want thats not already in `persistent`?
* What data should we get back from the app to store in `persistent`?
     * Structure of this data and intended/potential use.

**Comment by jonny983** [comment]
@ThePotatoGuy 
So I made a flowchart to show what a fully fleshed out version of my idea would be like so you know what I'm thinking of and that I'm serious about it.
[MASTravelAppFlowchart.pdf](https://github.com/Monika-After-Story/MonikaModDev/files/3588071/MASTravelAppFlowchart.pdf)

- This definitely isn't set in stone. 
- It shouldn't be implemented all at once.
- If you like it, we should start with some simple implementation to start.

It would take some work to make it good. If it's done right though, it definitely brings Monika closer to our reality and adds a whole new dynamic to the mod. 

I am well versed in Python and could help with the logic, but with no experience in Renpy though, it is probably best for others to handle that. 

- I plan on saving all data in some type of JSON or CSV file on the players phone, making it easy to read on the Python side

- To start, it will be location data with the associated activity and possibly a ranking of how
 much Monika enjoyed the experience.

- I'll format the data in the most convenient way possible.

Let me know what you think.

**Comment by ThePotatoGuy** [comment]
Yeah, seems good.

**Comment by Abc2557** [comment]
If this ever ends up being created and added, I'm gonna beg my parents to get me an Android phone or tablet haha

**Comment by lunulae** [comment]
If this is still in the works, I would love to volunteer any help I can give in writing dialogue and/or helping with visuals and UI. I have very little knowledge on the coding side but I would be excited to help out with making things look pretty.

**Reply by jonny983** [reply]
do you have a discord or something?

**Reply by multimokia** [reply]
We do have an official community discord if you're interested in chatting with us there over in the `#coding` channel, alternatively feel free to reach out to us by shooting us a DM.

Our discord is [here](https://discord.gg/monika-after-story)

**Comment by hussien0802** [comment]
is the app done? It's been a year. just asking :)

**Reply by jonny983** [reply]
haha no

**Reply by jonny983** [reply]
life got in the way. I'm thinking of starting it up again as I already did start on something small. I guess if there is still interest then I will be more inclined to pursue something.

**Comment by theoretischer** [comment]
I think that's the end of the app :(

**Reply by jonny983** [reply]
well, maybe not

**Reply by theoretischer** [reply]
**HE IS BACK**

**Reply by luckydog79** [reply]
I'm still interested

**Comment by TymNoir** [comment]
Hey, it's been 5 years. Are you abandoned the idea?
